requests-for-comments/2023-03-01-Temporal-RFC.md

```diff
@@ -0,0 +1,239 @@
+
+Finally, Temporal is a good fit because we are a small team. We should be focusing on the business of building our product. Nothing about what we are building that would be run on Temporal is novel enough to justify us to invest in a competing runtime.
+
+### Who uses it?
+
+- **Uber** (original developer - originally called [Cadence](https://github.com/uber/cadence))
+- **Coinbase**
+- **Doordash**
+- **Instacart**
+- **Hashicorp**
+- **Netflix**
+- **Snapchat**
+- **Stripe**
+- **Datadog**
+- **Box**
+- **Airbyte**
+- A whole lot more…
+
+### Quick Concept Overview
+
+A workflow in [Temporal.io](http://temporal.io/) is a set of simple, easy-to-understand steps that describe a complex process. Workflows are made up of activities, which are individual steps in the workflow that perform specific tasks. Activities can be executed in parallel, and Temporal provides retry logic, timeout logic, and exactly-once semantics. Workflows and activities can be composed together, and Temporal provides a dashboard for observing workflow status.
+
+# Example Workflow
+
+We currently use it for our manual backfills to S3 for certain clients. You can see an example workflow here:
+
+[https://github.com/posthog/energize](https://github.com/posthog/energize)
+
+Note: that this will be folded into the Posthog/Posthog repo. For now this is separate for example purposes.
+
+The specific workflow for S3 exporting can be found [here](https://github.com/PostHog/energize/blob/main/src/energize/workflows/exports/s3.py).
+
+# Temporal Cloud
+
+Why ☁️?
+
+There is one critical piece of the Temporal topology that causes loss if it goes down: The scheduler. As long as the scheduler is up. When you schedule a job - it will be completed. If the scheduler is down for an extended period of time there is a chance that the requesting service would drop the intent and that workflow would never be run. As long as the scheduler is up and we are with the tunable TTL of the workflow, as soon as a worker comes back online we will process that workflow.
+
+Luckily [Temporal.io](http://Temporal.io) has a hosted version of their scheduler where we can just submit our workflows and connect our workers to and the rest should be relatively smooth sailing. We only have to worry about writing the workflows themselves and keeping the workers happy.
+
+# Ownership
+
+This is why I’ve wanted to get you all here. Here is how ownership here would most likely shake out:
+
+**Infrastructure**
+
+- K8s Deployment of the workers that execute Workflows and Actions
+- CD of these workers
+
+**Billing**
+
+- Workflows and Actions written to be executed on Temporal workers
+
+**Pipeline**
+
+- Same, Workflows and actions written and scheduled to be executed on Temporal workers
+
+**Temporal cloud**
+
+- The scheduler itself. As long as the workflows are scheduled, they will be executed at some point in the future as long as a worker starts back up and the TTL does not expire the job.
```

Comment:
who owns Temporal cloud as a dependency management - their outages etc?

Reply:
That's a great question - I'd propose joint ownership between Infra and Pipelines.